src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/VsProjectAdapter.cs

```diff
@@ -220,9 +222,7 @@ public async Task<IEnumerable<(string ItemId, string[] ItemMetadata)>> GetBuildI
                 throw new ArgumentNullException(nameof(itemName));
             }
 
-            await _threadingService.JoinableTaskFactory.SwitchToMainThreadAsync();
-
-            var itemStorage = VsHierarchy as IVsBuildItemStorage;
+            var itemStorage = (IVsBuildItemStorage)VsHierarchy;
             if (itemStorage != null)
```

Comment:
This null check isn't necessary anymore if we are doing an explicit cast in the line above, now. We are asserting that VsHierarchy could never be null and it will always be castable to IVsBuildItemStorage. In that case, we can remove the null check for itemStorage.

Reply:
You actually caught a bug. 

The cast should be `as`. 
I'll fix that.